packages/scheduler/lib/models/tasks.integration.test.ts

```diff
@@ -0,0 +1,172 @@
+import { expect, describe, it, beforeAll, afterAll } from 'vitest';
+import { migrate } from '../db/migrate.js';
+import { clearDb } from '../db/test.helpers.js';
+import * as tasks from './tasks.js';
+import { taskStates } from '../types.js';
+import type { TaskState, Task } from '../types.js';
+
+describe('Task', () => {
+    beforeAll(async () => {
+        await migrate();
+    });
+
+    afterAll(async () => {
+        await clearDb();
+    });
+
+    it('should be successfully created', async () => {
+        const task = (
+            await tasks.create({
+                name: 'Test Task',
+                payload: { foo: 'bar' },
+                groupKey: 'groupA',
+                retryMax: 3,
+                retryCount: 1,
+                startsAfter: new Date(),
+                createdToStartedTimeoutSecs: 10,
+                startedToCompletedTimeoutSecs: 20,
+                heartbeatTimeoutSecs: 5
+            })
+        ).unwrap();
+        expect(task).toMatchObject({
+            id: expect.any(String) as string,
+            name: 'Test Task',
+            payload: { foo: 'bar' },
+            groupKey: 'groupA',
+            retryMax: 3,
+            retryCount: 1,
+            startsAfter: expect.any(Date) as Date,
```

Comment:
we could add a special expect with the correct type like I did in shared/lib/vitest.d.ts

Reply:
moved `vitest.d.ts` to utils and done what you mentioned